apis/k8ssandra/v1alpha1/k8ssandracluster_webhook.go

```diff
@@ -280,3 +290,30 @@ func (r *K8ssandraCluster) ValidateMedusa() error {
 
 	return nil
 }
+
+func (r *K8ssandraCluster) validateReaper() error {
+	if r.Spec.Reaper == nil {
+		return nil
+	}
+	if r.Spec.Reaper.StorageType == reaperapi.StorageTypeLocal && r.Spec.Reaper.StorageConfig == nil {
+		return ErrNoReaperStorageConfig
+	}
+	if r.Spec.Reaper.StorageType == reaperapi.StorageTypeLocal {
+		// we're checking for validity of the storage config in Reaper's webhook too, so this is a duplicate of that
+		// for now, I don't see a better way of reusing code to validate the storage config
+		// not checking StorageClassName because Kubernetes will use a default one if it's not set
+		if r.Spec.Reaper.StorageConfig.AccessModes == nil {
+			return ErrNoReaperAccessMode
+		}
+		if r.Spec.Reaper.StorageConfig.Resources.Requests == nil {
+			return ErrNoReaperResourceRequests
+		}
+		if r.Spec.Reaper.StorageConfig.Resources.Requests.Storage().IsZero() {
+			return ErrNoReaperStorageRequest
+		}
+	}
+	if r.Spec.Reaper.StorageType == reaperapi.StorageTypeLocal && r.Spec.Reaper.DeploymentMode == reaperapi.DeploymentModePerDc {
```

Comment:
I'm not sure if we need to enforce this here. Reaper won't come up, in this situation, so it's not the end of the world. On the other hand, it broke one e2e test, so perhaps it'll break stuff for users out there also.

Reply:
I've actually added a local storage mode to one of the fixtures in the StatefulSet PR [here](https://github.com/k8ssandra/k8ssandra-operator/pull/1365/files#diff-39d6b74cc9e0e10063db88b693f835c51d8f28e56cf1e8d3b9628d0496c520f7).  This is used by the `CreateReaperAndDatacenter` e2e test.